python/pyarrow/pandas_compat.py

```diff
@@ -730,15 +733,26 @@ def _reconstruct_block(item, columns=None, extension_columns=None):
             block_arr, categories=item['dictionary'],
             ordered=item['ordered'])
         block = _int.make_block(cat, placement=placement,
-                                klass=_int.CategoricalBlock)
+                                klass=_int.CategoricalBlock, ndim=ndim)
     elif 'timezone' in item:
         dtype = make_datetimetz(item['timezone'])
+        # TODO: once older pandas is dropped, use dtype.construct_array_type()
+        #  instead of hard-coding DatetimeArray
+        try:
+            from pandas.arrays import DatetimeArray
+            block_arr = DatetimeArray(block_arr, dtype=dtype)
+        except ImportError:
+            # older pandas versions
+            from pandas import DatetimeIndex
+            block_arr = DatetimeIndex(block_arr, dtype=dtype)
```

Comment:
This is also failing on older pandas. I assume because DatetimeArray and DatetimeIndex might be interpreting datetime64 differently? (wall time vs unix time? didn't check, but I remember discussions about it)